pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -231,6 +257,46 @@ func (c *Cluster) startProvisioning(config *provisionConfig) {
 		return
 	}
 
+	for _, volume := range c.ValidStorage.VolumeSources {
```

Comment:
For code readability of shorter methods how about putting this new code block in a separate method?

Reply:
cc @ashishranjan738

Reply:
can you review this again did some small changes on it.